errorreporting/src/main/java/com/example/errorreporting/QuickStart.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright 2017 Google Inc.
+ *
+ * <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * <p>http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.errorreporting;
+
+//[START errorreporting_quickstart]
+import com.google.cloud.ServiceOptions;
+import com.google.cloud.errorreporting.v1beta1.ReportErrorsServiceClient;
+import com.google.devtools.clouderrorreporting.v1beta1.ErrorContext;
+import com.google.devtools.clouderrorreporting.v1beta1.ProjectName;
+import com.google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent;
+import com.google.devtools.clouderrorreporting.v1beta1.SourceLocation;
+
+/**
+ * Snippet demonstrates using the Stackdriver Error Reporting API to report a custom error event.
+
+ * This library is not required on App Engine, errors written to stderr are automatically written
```

Comment:
@lesv PTAL if this suffices, I removed the stack trace reporting as well to make it clearer that using this API is an extreme rare case.